### DISCUSSION TITLE
Multi Tenancy for Development Teams

### DISCUSSION THREAD

**Discussion by disposab1e** [argoproj/argo-events#1275] [Q&A] [answered]
What is the recommended approach for multi tenancy? Is it possible or recommended to install dedicated namespaced Argo Events controllers for different development teams?

**Comment by whynowy** [comment]
Argo Events does support namespaced installation, but could you tell us what's the problem with cluster scope installation?

**Reply by disposab1e** [reply]
Actually with have dedicated ArgoCD installations for different Development Teams in the same cluster, so they can install, manage and use ArgoCD in a "team scope". Now we're thinking about adding the possibility to do the same with Argo Events, Workflows and Rollouts. Just searching for the best way to do it....  

@whynowy  Btw.. you climbed the nose?

**Comment by whynowy** [comment, marked as answer]
I wish ;)

I have no comments on the approach you did for argo-cd, but you can install Argo Events cluster wide, and different teams only need to take care of EventSource/Sensor/EventBus in their own namespace. But again, you can do namespaced install if you like. https://github.com/argoproj/argo-events/blob/master/docs/managed-namespace.md

**Reply by disposab1e** [reply]
@whynowy Thanks for your reply!

After experimenting a bit we decided to follow your recommendation for cluster wide installations for Workflow, Events and Rollouts.
